src/styles/calendar.scss

```
@@ -0,0 +1,44 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  width: $square;
+
+  @each $weekday, $number in $weekset {
+    &--start-day--#{$weekday} :first-child {
+      margin-left: #{($width + ($border * 2) + $gap) * $number};
+    }
+  }
+
+  @for $length from 28 through 31 {
+    &--month-length--#{$length} :nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
+
+  &__item {
+    display: flex;
+    width: $width;
+    height: $height;
+    border: $border solid black;
+    position: relative;
+    background-color: #eee;
+    cursor: pointer;
+    transition-duration: background-color 0.5s;
```

you should animate only properties you really need, not all
```suggestion
    transition: background-color 0.5s, transform 0.5s;
```
previous doesn't work, sorry for that(